### PR TITLE
[jupyter] add memory example to sql_db agent

### PR DESCRIPTION
**Pull request by skcoirz**
- Memory can largely enrich Agent convenience.
- This PR added an example showing how to use Memory in SQL DB Agent.
- Fixed a bug in the old SQLDatabaseToolkit initiation case as `llm` is a required input in the latest version.
- Re-tested several old examples to confirm everything works the same.

### CODE REVIEW

File: docs/modules/agents/toolkits/examples/sql_database.ipynb [hunk 1]
```diff
@@ -501,13 +514,312 @@
...
     "agent_executor.run(\"Who are the top 3 best selling artists?\")"
    ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "8feee2be",
+   "metadata": {},
+   "source": [
+    "# Memory Use Case Example\n",
```
Comment (hwchase17) [inline docs/modules/agents/toolkits/examples/sql_database.ipynb:524]: this should be `##` or it will show up in the sidebar

**Review by hwchase17 — COMMENTED** [review]
minor nit, otherwise lgtm!

File: docs/modules/agents/toolkits/examples/sql_database.ipynb (hunk 1 above)
Comment (skcoirz) [reply docs/modules/agents/toolkits/examples/sql_database.ipynb:524]: nah, i see. fixing now!

**Comment by dev2049** [comment]
should we be using gpt-4 in example? probably better to use something anyone can access (unless the example doesn't work with anything else)

**Comment by skcoirz** [comment]
> should we be using gpt-4 in example? probably better to use something anyone can access (unless the example doesn't work with anything else)

sure, have updated to use gpt-3.5-turbo

**Review by hwchase17 — COMMENTED** [review]
since using chat model, should use ChatOpenAI

Otherwise looks good

**Comment by graemeglass** [comment]
> > should we be using gpt-4 in example? probably better to use something anyone can access (unless the example doesn't work with anything else)
> 
> sure, have updated to use gpt-3.5-turbo

Since this PR hasn't been merged yet, you could change it back to gpt-4. 
It became publicly available on [6 July 2023](https://openai.com/blog/gpt-4-api-general-availability).

**Comment by leo-gan** [comment]
@skcoirz Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that, ping me and I push this PR for the review. Thanks!

**Pull request closed without merging**